File: app/src/main/java/com/example/steaminvestmentmanager/utilclasses/PriceURLCreator.java
package com.example.steaminvestmentmanager.utilclasses;

import java.io.UnsupportedEncodingException;
import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;

public class PriceURLCreator {
    public static String getURL(SteamItem steamItem, boolean isUserCurrency) {
        String itemPriceoverviewURLBase = "https://steamcommunity.com/market/priceoverview/?";
        itemPriceoverviewURLBase += "appid=" + steamItem.getAppID();
        if (isUserCurrency) {
            itemPriceoverviewURLBase += "&currency=" + CurrencyData.getCurrency();
        }else {
            itemPriceoverviewURLBase += "&currency=" + steamItem.getCurrentCurrency();
        }
        String market_hash_name_encoded;
        try {
            market_hash_name_encoded = URLEncoder.encode(steamItem.getMarket_hash_name(), String.valueOf(StandardCharsets.UTF_8));
        } catch (UnsupportedEncodingException e) {
            e.printStackTrace();
            return "";
        }
        itemPriceoverviewURLBase += "&market_hash_name=" + market_hash_name_encoded;
        return itemPriceoverviewURLBase;
    }
}

File: app/src/main/java/com/example/steaminvestmentmanager/utilclasses/PriceURLCreator.kt
package com.example.steaminvestmentmanager.utilclasses

import java.io.UnsupportedEncodingException
import java.net.URLEncoder
import java.nio.charset.StandardCharsets

object PriceURLCreator {
    fun getURL(steamItem: SteamItem, isUserCurrency: Boolean): String {
        var itemPriceoverviewURLBase = "https://steamcommunity.com/market/priceoverview/?"
        itemPriceoverviewURLBase += "appid=" + steamItem.appID
        itemPriceoverviewURLBase += if (isUserCurrency) {
            "&currency=" + CurrencyData.getCurrency()
        } else {
            "&currency=" + steamItem.currentCurrency
        }
        val market_hash_name_encoded: String = try {
            URLEncoder.encode(steamItem.market_hash_name, StandardCharsets.UTF_8.toString())
        } catch (e: UnsupportedEncodingException) {
            e.printStackTrace()
            return ""
        }
        itemPriceoverviewURLBase += "&market_hash_name=$market_hash_name_encoded"
        return itemPriceoverviewURLBase
    }
}